x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSingleNodeTests.java

```diff
@@ -446,9 +536,22 @@ private ThreadContext prepareThreadContext(SignedJWT signedJWT, String clientSec
         if (clientSecret != null) {
             threadContext.putHeader(
                 JwtRealm.HEADER_CLIENT_AUTHENTICATION,
-                JwtRealm.HEADER_SHARED_SECRET_AUTHENTICATION_SCHEME + " " + clientSecret
+                JwtRealmSettings.HEADER_SHARED_SECRET_AUTHENTICATION_SCHEME + " " + clientSecret
             );
         }
         return threadContext;
     }
+
+    private static GrantApiKeyRequest getGrantApiKeyForJWT(SignedJWT signedJWT, String sharedSecret) {
+        GrantApiKeyRequest grantApiKeyRequest = new GrantApiKeyRequest();
+        grantApiKeyRequest.getGrant().setType("access_token");
+        grantApiKeyRequest.getGrant().setAccessToken(new SecureString(signedJWT.serialize().toCharArray()));
+        if (sharedSecret != null) {
+            grantApiKeyRequest.getGrant()
+                .setClientAuthentication(new Grant.ClientAuthentication("SharedSecret", new SecureString(sharedSecret.toCharArray())));
```

Comment:
I intentionally used the verbatim value "SharedSecret" so that the test will fail if we ever change `JwtRealmSettings.HEADER_SHARED_SECRET_AUTHENTICATION_SCHEME` . Basically the test failure tells you that clients in the wild rely on the value of that variable so it's not easily changeable (to constrast with other public static variables that can more easily be changed).